sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/HttpHelper.cs

```diff
@@ -20,136 +15,81 @@ internal static class HttpHelper
         /// Reference: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#http-client.
         /// </summary>
         /// <param name="tagObjects">Activity Tags</param>
-        /// <returns></returns>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static string GetUrl(this AzMonList tagObjects)
+        /// <param name="url">URL</param>
+        /// <param name="urlAuthority">Host name or IP address and the port number </param>
+        internal static void GenerateUrlAndAuthority(this AzMonList tagObjects, out string url, out string urlAuthority)
         {
+            urlAuthority = null;
+            url = null;
+
             var httpurl = tagObjects.GetTagValue(SemanticConventions.AttributeHttpUrl);
 
             if (httpurl != null && Uri.TryCreate(httpurl.ToString(), UriKind.RelativeOrAbsolute, out var uri) && uri.IsAbsoluteUri)
             {
-                return uri.AbsoluteUri;
+                url = uri.AbsoluteUri;
+                urlAuthority = uri.Authority;
+                return;
             }
 
-            string url = null;
-
-            var httpScheme = tagObjects.GetTagValue(SemanticConventions.AttributeHttpScheme);
+            var httpScheme = tagObjects.GetTagValue(SemanticConventions.AttributeHttpScheme)?.ToString();
 
-            if (httpScheme != null)
+            if (!string.IsNullOrWhiteSpace(httpScheme))
             {
-                var httpHost = tagObjects.GetTagValue(SemanticConventions.AttributeHttpHost);
+                var httpHost = tagObjects.GetTagValue(SemanticConventions.AttributeHttpHost)?.ToString();
 
-                if (httpHost != null)
+                if (!string.IsNullOrWhiteSpace(httpHost))
                 {
                     var httpPortAndTarget = tagObjects.GetTagValues(SemanticConventions.AttributeHttpHostPort, SemanticConventions.AttributeHttpTarget);
 
                     if (httpPortAndTarget[0] != null && httpPortAndTarget[0].ToString() != "80" && httpPortAndTarget[0].ToString() != "443")
                     {
-                        url = $"{httpScheme}://{httpHost}:{httpPortAndTarget[0]}{httpPortAndTarget[1]}";
+                        var IsColon = (string.IsNullOrWhiteSpace(httpPortAndTarget[0]?.ToString()) ? null : Colon);
```

Comment:
The null check and `ToString()` part would probably require some clean up (e.g. in the if-condition we already made the validation that `httpPortAndTarget[0] != null`.